targets/TARGET_STM/TARGET_STM32F0/flash_api.c

```diff
@@ -93,7 +74,9 @@ int32_t flash_erase_sector(flash_t *obj, uint32_t address)
         status = -1;
     }
 
-    flash_lock();
+	core_util_critical_section_exit();
+
+	HAL_FLASH_Lock();
```

Comment:
I would prefer to align code with HAL_FLASH_Unlock call:

    if (HAL_FLASH_Lock() != HAL_OK) {
        return -1;
    }

 ?

Reply:
Good idea.